src/sep6/DepositProvider.ts

```diff
@@ -0,0 +1,65 @@
+import queryString from "query-string";
+
+import { Omit } from "../util";
+import { fetchKycInBrowser } from "./fetchKycInBrowser";
+import { TransferProvider } from "./TransferProvider";
+import {
+  DepositRequest,
+  FeeArgs,
+  InteractiveKycNeeded,
+  TransferResponse,
+} from "./types";
+
+export class DepositProvider extends TransferProvider {
+  public async deposit(args: DepositRequest): Promise<TransferResponse> {
+    const search = queryString.stringify(args);
+    const response = await fetch(`${this.transferServer}/deposit?${search}`);
+    return response.json();
+  }
+
+  public async fetchSupportedAssets() {
+    const { deposit } = await this.fetchInfo();
+    return deposit;
+  }
+
+  public async fetchFinalFee(args: Omit<FeeArgs, "operation">) {
+    return super.fetchFinalFee({
+      ...args,
+      operation: "deposit",
+    });
+  }
+
+  public async fetchKycInBrowser({
+    response,
+    request,
+    window: windowContext,
+  }: {
+    response: InteractiveKycNeeded;
+    request: DepositRequest;
+    window: Window;
```

Comment:
You may want to check what type React Native's webview packages return?

https://facebook.github.io/react-native/docs/webview

Reply:
I looked into that briefly, react-native appears to use a non-standard PostMessage that I'm not sure we should specifically target. I'm thinking we leave it  for now and circle back once we can dogfood/get feedback from a react-native use case.

Though I do see that I forgot to add the original request to the callback URL like we'd talked about, I'll add that.